src/librustdoc/html/static/main.js

```diff
@@ -664,18 +663,15 @@ function defocusSearchBar() {
                 results = {}, results_in_args = {}, results_returned = {},
                 split = valLower.split("::");
 
-            var length = split.length;
-            for (var z = 0; z < length; ++z) {
+            for (var z = split.length - 1; z >= 0; --z) {
                 if (split[z] === "") {
                     split.splice(z, 1);
-                    z -= 1;
                 }
             }
```

Comment:
```suggestion
split = split.filter(function(segment) { return segment !== ""; });
```

Reply:
That is a **huge** speed gain. (84% apparently). Great catch!